tools/codegen/api/python.py

```diff
@@ -173,6 +173,53 @@
 #      }
 #
 
+# TODO: stick this more firmly in the data model somewhere?
+def namedtuple_fieldnames(returns: Tuple[Return]) -> List[str]:
+    if len(returns) <= 1 or all(map(lambda r: r.name is None, returns)):
+        return []
+    else:
+        if any(map(lambda r: r.name is None, returns)):
+            # When building on Windows, `PyStructSequence_UnnamedField` could not be
+            # resolved by the linker for some reason, which cause error in building:
+            #
+            # python_nn_functions.cpp.obj : error LNK2001: unresolved external symbol
+            # PyStructSequence_UnnamedField
+            #
+            # Thus, at this point in time, we do not support unnamed
+            # fields in namedtuple; you must either name all fields,
+            # or none of them.
+            raise ValueError("Unnamed field is not supported by codegen")
+
+        return list(map(lambda r: str(r.name), returns))
+
+@dataclass(frozen=True)
+class PythonReturns:
```

Comment:
@ljk53 Thinking aloud: now that we have more structure in the main model, I'm wondering what the delta between the model and the Python variants are. In particular, if the main model has enough structure (now) to conveniently do direct translation, we can dispense with the Python classes here entirely (@bdhirsh another reason I didn't really want us to go and copy paste the Python classes). If it doesn't have enough information, maybe we should add more stuff to it!

(also cc @smessmer )

Reply:
(@bdhirsh no action necessary on this PR)